### PR TITLE
fixed #76316, don't delete hnsendpoint if it's referenced by another service

### LINKED ISSUE #76316: winkernel proxy is deleting hnsendpoint while it's still referenced by another k8s service.
**What happened**: We have two service(A & B) pointing to the same pod and different port, if we delete service A, service B becomes unaccessible from nodes other than the node pod is running at.

**What you expected to happen**: Deleting service A doesn't affect servicie B.

**How to reproduce it (as minimally and precisely as possible)**: Create two service pointing to the same pod and different port, check service accessibility from other nodes, then delete service A, check service B accessibility from other nodes.

**Anything else we need to know?**:  winkernel proxy refers hnsEndpoint from two endpointInfo structure if they points to same ip, both endpointInfo has refCount 1. If service A deleted, winkernel proxy will delete hnsEndpoint, causing another endpointInfo referring invalid  hnsEndpoint.

**Environment**:
- Kubernetes version (use `kubectl version`):  1.12.4
- Cloud provider or hardware configuration:  aliyun
- OS (e.g: `cat /etc/os-release`): windows 1809
- Kernel (e.g. `uname -a`): 
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by jxiewei**
fixed #76316, don't delete hnsendpoint if it's referenced by another service

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
/kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #76316

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Review by feiskyer — COMMENTED** [review]
/ok-to-test

File: pkg/proxy/winkernel/proxier.go [hunk 1]
```diff
@@ -182,12 +182,14 @@ func (ep *endpointsInfo) Cleanup() {
 	// Never delete a Local Endpoint. Local Endpoints are already created by other entities.
 	// Remove only remote endpoints created by this service
 	if ep.refCount <= 0 && !ep.isLocal {
```
Comment (feiskyer) [inline pkg/proxy/winkernel/proxier.go]: so ep.refCount doesn't catch such cases?

Comment (jxiewei) [reply pkg/proxy/winkernel/proxier.go]: Can't, two different ep are  created in such case, all of them have refCount 1.
Deleting anyone of them will trigger deletion of hnsEdnpoint.

**Comment by ksubrmnn** [comment]
Thank you for your PR! Looks like this issue is also being addressed by https://github.com/kubernetes/kubernetes/pull/76406 

I prefer the approach in 76406 since it avoids the issue by not creating duplicate EpInfos. I’m going to validate 76406 this week and merge that.

**Pull request closed without merging**